nightfall-optimist/src/services/check-block.mjs

```diff
@@ -80,10 +80,17 @@ async function checkBlock(block, transactions) {
   // to check a block against itself (hence the second filter).
   const storedMinedNullifiers = await retrieveMinedNullifiers(); // List of Nullifiers stored by blockProposer
   const blockNullifiers = transactions.map(tNull => tNull.nullifiers).flat(Infinity); // List of Nullifiers in block
+  const transactionNullifiers = blockNullifiers.filter(
```

Comment:
Because we re checking `assert(nullifier[0].hash != nullifier[1].hash)` inside `double_transfer.zok`, if same commitment is spent twice in a double transfer, the proof verification will. This will be caught by `checkBlock` under `verifyProof` of `checkTransaction`. So this is redundant fix here